Wikipedia/Code/WMFFeedContentSource.m

```diff
@@ -390,18 +390,36 @@ - (void)scheduleNotificationsForFeedDay:(WMFFeedDayResponse *)feedDay onDate:(NS
     }
 
     WMFFeedNewsStory *newsStory = feedDay.newsStories.firstObject;
-
     if (!newsStory) {
         done();
         return;
     }
-
+    
+    NSDate *newsDate = newsStory.midnightUTCMonthAndDay;
+    if (newsDate) {
+        NSCalendar *utcCalendar = [NSCalendar wmf_utcGregorianCalendar];
+        NSDate *midnightUTCDate = date.wmf_midnightUTCDateFromLocalDate;
+        NSDateComponents *nowComponents = [utcCalendar components:NSCalendarUnitYear | NSCalendarUnitMonth | NSCalendarUnitDay fromDate:midnightUTCDate];
+        NSDateComponents *newsComponents = [utcCalendar components:NSCalendarUnitYear | NSCalendarUnitMonth | NSCalendarUnitDay fromDate:newsDate];
+        NSInteger year = nowComponents.year;
+        if (nowComponents.month == 1 && newsComponents.month == 12) {
+            year--;
+        }
+        newsComponents.year = year;
+        newsDate = [utcCalendar dateFromComponents:newsComponents];
+        if (newsDate && midnightUTCDate && [utcCalendar wmf_daysFromDate:newsDate toDate:midnightUTCDate] > 1) {
```

Comment:
I abstracted this into an extension on calendar. It's not a boolean, but it's clearer.